Library/Homebrew/cask/download.rb

```diff
@@ -51,9 +53,17 @@ def version
     def fetch(quiet: nil, verify_download_integrity: true, timeout: nil)
       downloader.quiet! if quiet
 
+      @urls_index = 0
       begin
         super(verify_download_integrity: false, timeout: timeout)
       rescue DownloadError => e
+        @urls_index += 1
+        if @urls_index < @urls.length
+          @downloader = nil
+          puts "Trying a mirror..."
+          retry
+        end
+        @urls_index = 0
```

Comment:
Some oddness to try to allow mirrors have their own specs.

`Downloadable` has mirrors-support for URL strings which will use the same specs as main URL. I guess the simpler option would be to only support `verified:` and then set `@mirrors = ...`